src/encoded/tests/test_upgrade_analyte.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any, Dict
+
+import pytest
+from pyramid.router import Router
+
+from .utils import get_upgrader
+
+
+@pytest.mark.parametrize(
+    "analyte,expected",
+    [
+        ({}, {"schema_version": "2"}),
+        (
+            {"components": [], "schema_version": "1"},
+            {"schema_version": "2"},
+        ),
+        (
+            {"components": ["Genomic DNA"], "schema_version": "1"},
+            {"molecule_detail": ["Genomic DNA"], "schema_version": "2"},
```

Comment:
You might consider testing a length 2 array though is probably not necessary given how basic the upgrade is.